modules/javafx.graphics/src/main/java/javafx/css/SimpleSelector.java

```diff
@@ -306,7 +361,7 @@ private boolean matchStyleClasses(StyleClassSet otherStyleClasses) {
         if ((this.id == null) ? (other.id != null) : !this.id.equals(other.id)) {
             return false;
         }
-        if (this.styleClassSet.equals(other.styleClassSet) == false) {
+        if (this.selectorStyleClassNames.equals(other.selectorStyleClassNames) == false) {
```

Comment:
Maybe replace the `==false` with `!`.

Reply:
I've kept the style the same, there is another line like this just below it.  Don't like it either personally :)